### ISSUE TITLE
Possible problems with Math::BigInt::Calc

### ISSUE THREAD

**Issue by merkys** [Materials-Consortia/OPTiMaDe-Filter#2]
When using bignum in JSON reading, generated filters contain Math::BigInt::Calc objects for values:
```Perl
>                                  'filter' => bless( {
>                                                       'operands' => [
>                                                                       bless( {
>                                                                                'name' => [
>                                                                                            'bibliography',
>                                                                                            'id'
>                                                                                          ]
>                                                                              }, 'OPTIMADE::Filter::Property' ),
>                                                                       bless( {
>                                                                                '_es' => '+',
>                                                                                '_e' => bless( [
>                                                                                                 0
>                                                                                               ], 'Math::BigInt::Calc' ),
>                                                                                'sign' => '+',
>                                                                                '_m' => bless( [
>                                                                                                 1
>                                                                                               ], 'Math::BigInt::Calc' )
>                                                                              }, 'Math::BigFloat' )
>                                                                     ],
>                                                       'operator' => '='
>                                                     }, 'OPTIMADE::Filter::Comparison' )
```
These might not be correctly converted into values for Perl DBI inclusion.

**Comment by merkys** [comment]
This is the issue with upstream code. `OPTIMADE::Filter` does not change values passed to it during filter construction, as demonstrated in `t/07bigint.t`.

**Comment by merkys** [comment]
Actually, `modify()` subroutine does affect objects that have `modify()` methods themselves. This has been fixed in d0b76e24c1f18e0bab452c2e9d65ad38f61f113f.
